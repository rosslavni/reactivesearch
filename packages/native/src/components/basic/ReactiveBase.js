import React, { Component } from 'react';
import { Provider } from 'react-redux';
import { Container } from 'native-base';
import Appbase from 'appbase-js';

import configureStore from '@appbaseio/reactivecore';

import types from '@appbaseio/reactivecore/lib/utils/types';

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : '*';

		const credentials = props.url && props.url.trim() !== '' && !props.credentials
			? 'user:pass'
			: props.credentials;

		const config = {
			url: props.url && props.url.trim() !== '' ? props.url : 'https://scalr.api.appbase.io',
			app: props.app,
			credentials,
			type: this.type,
		};

		const appbaseRef = new Appbase(config);
		this.store = configureStore({ config, appbaseRef });
	}

	render() {
		return (
			<Provider store={this.store}>
				<Container>
					{this.props.children}
				</Container>
			</Provider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.string,
	url: types.string,
	credentials: types.string,
	app: types.stringRequired,
	children: types.children,
};

export default ReactiveBase;
