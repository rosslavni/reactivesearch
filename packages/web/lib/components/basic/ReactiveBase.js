var _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; };

function _classCallCheck(instance, Constructor) { if (!(instance instanceof Constructor)) { throw new TypeError("Cannot call a class as a function"); } }

function _possibleConstructorReturn(self, call) { if (!self) { throw new ReferenceError("this hasn't been initialised - super() hasn't been called"); } return call && (typeof call === "object" || typeof call === "function") ? call : self; }

function _inherits(subClass, superClass) { if (typeof superClass !== "function" && superClass !== null) { throw new TypeError("Super expression must either be null or a function, not " + typeof superClass); } subClass.prototype = Object.create(superClass && superClass.prototype, { constructor: { value: subClass, enumerable: false, writable: true, configurable: true } }); if (superClass) Object.setPrototypeOf ? Object.setPrototypeOf(subClass, superClass) : subClass.__proto__ = superClass; }

import React, { Component } from "react";
import { Provider } from "react-redux";
import Appbase from "appbase-js";

import configureStore from "@appbaseio/reactivecore";
import types from "@appbaseio/reactivecore/lib/utils/types";
import URLParamsProvider from "./URLParamsProvider";

var URLSearchParams = require("url-search-params");

var ReactiveBase = function (_Component) {
	_inherits(ReactiveBase, _Component);

	function ReactiveBase(props) {
		_classCallCheck(this, ReactiveBase);

		var _this = _possibleConstructorReturn(this, _Component.call(this, props));

		_this.type = props.type ? props.type : "*";

		var credentials = props.url && props.url.trim() !== "" && !props.credentials ? "user:pass" : props.credentials;

		var config = {
			url: props.url && props.url.trim() !== "" ? props.url : "https://scalr.api.appbase.io",
			app: props.app,
			credentials: props.credentials,
			type: _this.type
		};

		_this.params = new URLSearchParams(window.location.search);
		var selectedValues = {};

		try {
			for (var _iterator = _this.params.keys(), _isArray = Array.isArray(_iterator), _i = 0, _iterator = _isArray ? _iterator : _iterator[Symbol.iterator]();;) {
				var _extends2;

				var _ref;

				if (_isArray) {
					if (_i >= _iterator.length) break;
					_ref = _iterator[_i++];
				} else {
					_i = _iterator.next();
					if (_i.done) break;
					_ref = _i.value;
				}

				var key = _ref;

				selectedValues = _extends({}, selectedValues, (_extends2 = {}, _extends2[key] = { value: JSON.parse(_this.params.get(key)) }, _extends2));
			}
		} catch (e) {
			selectedValues = {};
		};

		var appbaseRef = new Appbase(config);
		_this.store = configureStore({ config: config, appbaseRef: appbaseRef, selectedValues: selectedValues });
		return _this;
	}

	ReactiveBase.prototype.render = function render() {
		return React.createElement(
			Provider,
			{ store: this.store },
			React.createElement(
				URLParamsProvider,
				{ params: this.params },
				this.props.children
			)
		);
	};

	return ReactiveBase;
}(Component);

ReactiveBase.propTypes = {
	type: types.type,
	url: types.url,
	credentials: types.credentials,
	app: types.app,
	children: types.children
};

export default ReactiveBase;