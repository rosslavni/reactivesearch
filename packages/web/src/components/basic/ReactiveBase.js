import React, { Component } from "react";
import { Provider } from "react-redux";
import Appbase from "appbase-js";

import configureStore from "@appbaseio/reactivecore";
import types from "@appbaseio/reactivecore/lib/utils/types";

import { base } from "../../styles";

class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : "*";

		const credentials = props.url && props.url.trim() !== "" && !props.credentials
			? "user:pass"
			: props.credentials;

		const config = {
			url: props.url && props.url.trim() !== "" ? props.url : "https://scalr.api.appbase.io",
			app: props.app,
			credentials: props.credentials,
			type: this.type
		};

		const appbaseRef = new Appbase(config);
		this.store = configureStore({ config, appbaseRef });
	}

	render() {
		return (
			<Provider store={this.store}>
				<div className={base}>
					{this.props.children}
				</div>
			</Provider>
		);
	}
}

ReactiveBase.propTypes = {
	type: types.type,
	url: types.url,
	credentials: types.credentials,
	app: types.app,
	children: types.children
}
export default ReactiveBase;
