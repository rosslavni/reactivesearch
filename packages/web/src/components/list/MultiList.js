import React, { Component } from "react";
import { connect } from "react-redux";

import {
	addComponent,
	removeComponent,
	watchComponent,
	updateQuery,
	setQueryOptions
} from "@appbaseio/reactivecore/lib/actions";
import {
	isEqual,
	getQueryOptions,
	pushToAndClause,
	checkValueChange,
	getAggsOrder,
	checkPropChange,
	checkSomePropChange
} from "@appbaseio/reactivecore/lib/utils/helper";

import types from "@appbaseio/reactivecore/lib/utils/types";

import Title from "../../styles/Title";
import Input from "../../styles/Input";
import { UL, Checkbox } from "../../styles/FormControlList";

class MultiList extends Component {
	constructor(props) {
		super(props);

		this.state = {
			currentValue: {},
			options: [],
			searchTerm: ""
		};
		this.internalComponent = props.componentId + "__internal";
	}

	componentWillMount() {
		this.props.addComponent(this.internalComponent);
		this.props.addComponent(this.props.componentId);
		this.setReact(this.props);

		this.updateQueryOptions(this.props);

		if (this.props.selectedValue) {
			this.setValue(this.props.selectedValue, true);
		} else if (this.props.defaultSelected) {
			this.setValue(this.props.defaultSelected, true);
		}
	}

	componentWillReceiveProps(nextProps) {
		checkPropChange(
			this.props.react,
			nextProps.react,
			() => this.setReact(nextProps)
		);
		checkPropChange(
			this.props.options,
			nextProps.options,
			() => {
				this.setState({
					options: nextProps.options[nextProps.dataField].buckets || []
				});
			}
		);
		checkSomePropChange(
			this.props,
			nextProps,
			["size", "sortBy"],
			() => this.updateQueryOptions(nextProps)
		);

		let selectedValue = Object.keys(this.state.currentValue);

		if (this.props.selectAllLabel) {
			selectedValue = selectedValue.filter(val => val !== this.props.selectAllLabel);

			if (!!this.state.currentValue[this.props.selectAllLabel]) {
				selectedValue = [this.props.selectAllLabel];
			}
		}

		if (this.props.defaultSelected !== nextProps.defaultSelected) {
			this.setValue(nextProps.defaultSelected, true);
		} else if (!isEqual(selectedValue, nextProps.selectedValue)) {
			this.setValue(nextProps.selectedValue, true);
		}
	}

	componentWillUnmount() {
		this.props.removeComponent(this.props.componentId);
		this.props.removeComponent(this.internalComponent);
	}

	setReact = (props) => {
		const { react } = props;
		if (react) {
			const newReact = pushToAndClause(react, this.internalComponent);
			props.watchComponent(props.componentId, newReact);
		} else {
			props.watchComponent(props.componentId, { and: this.internalComponent });
		}
	};

	defaultQuery = (value, props) => {
		let query = null;
		const type = props.queryFormat === "or" ? "terms" : "term";
		if (this.props.selectAllLabel && value.includes(this.props.selectAllLabel)) {
			query = {
				exists: {
					field: props.dataField
				}
			};
		} else if (value) {
			let listQuery;
			if (props.queryFormat === "or") {
				listQuery = {
					[type]: {
						[props.dataField]: value
					}
				};
			} else {
				// adds a sub-query with must as an array of objects for each term/value
				const queryArray = value.map(item => (
					{
						[type]: {
							[props.dataField]: item
						}
					}
				));
				listQuery = {
					bool: {
						must: queryArray
					}
				};
			}

			query = value.length ? listQuery : null;
		}
		return query;
	};

	setValue = (value, isDefaultValue = false, props = this.props) => {
		const { selectAllLabel } = this.props;
		let { currentValue } = this.state;
		let finalValues = null;

		if (selectAllLabel &&
			((Array.isArray(value) && value.includes(selectAllLabel)) ||
			(typeof value === "string" && value === selectAllLabel))) {
			if (!!currentValue[selectAllLabel]) {
				currentValue = {};
				finalValues = [];
			} else {
				this.state.options.forEach(item => {
					currentValue[item.key] = true;
				});
				currentValue[selectAllLabel] = true;
				finalValues = [selectAllLabel];
			}
		} else if (isDefaultValue) {
			finalValues = value;
			currentValue = {};
			value && value.forEach(item => {
				currentValue[item] = true;
			});

			if (selectAllLabel && selectAllLabel in currentValue) {
				const { [selectAllLabel]: del, ...obj } = currentValue;
				currentValue = { ...obj };
			}
		} else {
			if (currentValue[value]) {
				const { [value]: del, ...rest } = currentValue;
				currentValue = { ...rest };
			} else {
				currentValue[value] = true;
			}

			if (selectAllLabel && selectAllLabel in currentValue) {
				const { [selectAllLabel]: del, ...obj } = currentValue;
				currentValue = { ...obj };
			}
			finalValues = Object.keys(currentValue);
		}

		const performUpdate = () => {
			this.setState({
				currentValue
			}, () => {
				this.updateQuery(finalValues, props);
			});
		}

		checkValueChange(
			props.componentId,
			finalValues,
			props.beforeValueChange,
			props.onValueChange,
			performUpdate
		);
	};

	updateQuery = (value, props) => {
		const query = props.customQuery || this.defaultQuery;
		let onQueryChange = null;
		if (props.onQueryChange) {
			onQueryChange = props.onQueryChange;
		}
		props.updateQuery({
			componentId: props.componentId,
			query: query(value, props),
			value,
			label: props.filterLabel,
			showFilter: props.showFilter,
			onQueryChange,
			URLParams: props.URLParams
		});
	};

	updateQueryOptions = (props) => {
		const queryOptions = getQueryOptions(props);
		queryOptions.aggs = {
			[props.dataField]: {
				terms: {
					field: props.dataField,
					size: props.size,
					order: getAggsOrder(props.sortBy)
				}
			}
		}
		props.setQueryOptions(this.internalComponent, queryOptions);
		// Since the queryOptions are attached to the internal component,
		// we need to notify the subscriber (parent component)
		// that the query has changed because no new query will be
		// auto-generated for the internal component as its
		// dependency tree is empty
		props.updateQuery({
			componentId: this.internalComponent,
			query: null
		});
	};

	handleInputChange = (e) => {
		const { value } = e.target;
		this.setState({
			searchTerm: value
		});
	};

	renderSearch = () => {
		if (this.props.showSearch) {
			return <Input
				onChange={this.handleInputChange}
				value={this.state.searchTerm}
				placeholder={this.props.placeholder}
				style={{
					margin: "0 0 8px"
				}}
			/>
		}
		return null;
	};

	render() {
		const { selectAllLabel } = this.props;

		if (this.state.options.length === 0) {
			return null;
		}

		return (
			<div style={this.props.style} className={this.props.className}>
				{this.props.title && <Title>{this.props.title}</Title>}
				{this.renderSearch()}
				<UL>
					{
						selectAllLabel
							? (<li key={selectAllLabel}>
								<Checkbox
									id={selectAllLabel}
									name={selectAllLabel}
									value={selectAllLabel}
									onClick={e => this.setValue(e.target.value)}
									checked={!!this.state.currentValue[selectAllLabel]}
									show={this.props.showCheckbox}
								/>
								<label htmlFor={selectAllLabel}>
									{selectAllLabel}
								</label>
							</li>)
							: null
					}
					{
						this.state.options
							.filter(item => {
								if (this.props.showSearch && this.state.searchTerm) {
									return item.key.toLowerCase().includes(this.state.searchTerm.toLowerCase());
								}
								return true;
							})
							.map(item => (
								<li key={item.key}>
									<Checkbox
										id={item.key}
										name={this.props.componentId}
										value={item.key}
										onClick={e => this.setValue(e.target.value)}
										checked={!!this.state.currentValue[item.key]}
										onChange={() => {}}
										show={this.props.showCheckbox}
									/>
									<label htmlFor={item.key}>
										{item.key}
										{
											this.props.showCount &&
											` (${item.doc_count})`
										}
									</label>
								</li>
							))
					}
				</UL>
			</div>
		);
	}
}

MultiList.propTypes = {
	componentId: types.stringRequired,
	addComponent: types.funcRequired,
	dataField: types.stringRequired,
	sortBy: types.sortByWithCount,
	setQueryOptions: types.funcRequired,
	updateQuery: types.funcRequired,
	defaultSelected: types.stringArray,
	react: types.react,
	options: types.options,
	removeComponent: types.funcRequired,
	beforeValueChange: types.func,
	onValueChange: types.func,
	customQuery: types.func,
	onQueryChange: types.func,
	placeholder: types.string,
	title: types.title,
	showCheckbox: types.boolRequired,
	filterLabel: types.string,
	selectedValue: types.selectedValue,
	queryFormat: types.queryFormatSearch,
	URLParams: types.boolRequired,
	showCount: types.bool,
	size: types.number,
	showSearch: types.bool,
	selectAllLabel: types.string,
	style: types.style,
	className: types.string
}

MultiList.defaultProps = {
	size: 100,
	sortBy: "count",
	showCheckbox: true,
	queryFormat: "or",
	URLParams: false,
	showCount: true,
	placeholder: "Search",
	showSearch: true,
	style: {},
	className: null
}

const mapStateToProps = (state, props) => ({
	options: state.aggregations[props.componentId],
	selectedValue: state.selectedValues[props.componentId] && state.selectedValues[props.componentId].value || []
});

const mapDispatchtoProps = dispatch => ({
	addComponent: component => dispatch(addComponent(component)),
	removeComponent: component => dispatch(removeComponent(component)),
	watchComponent: (component, react) => dispatch(watchComponent(component, react)),
	updateQuery: (updateQueryObject) => dispatch(updateQuery(updateQueryObject)),
	setQueryOptions: (component, props) => dispatch(setQueryOptions(component, props))
});

export default connect(mapStateToProps, mapDispatchtoProps)(MultiList);
