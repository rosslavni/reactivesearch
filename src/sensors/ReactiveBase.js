import React, { Component } from "react";
import { Provider } from "react-redux";
import { View } from "react-native";

import configureStore from "../store/configureStore";

export default class ReactiveBase extends Component {
	constructor(props) {
		super(props);

		this.type = props.type ? props.type : "*";
		const config = {
			url: "scalr.api.appbase.io",
			app: props.app,
			credentials: props.credentials,
			type: this.type
		};
		this.store = configureStore({config});
	}

	render() {
		return (
			<Provider store={this.store}>
				<View style={this.props.style}>
					{this.props.children}
				</View>
			</Provider>
		);
	}
}
