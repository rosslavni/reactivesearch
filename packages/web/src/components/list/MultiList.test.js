import React from 'react';
import renderer from 'react-test-renderer';
import MultiList from './MultiList';
import ReactiveBase from '../basic/ReactiveBase';

it('should render no results message', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<MultiList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					renderNoResults={() => 'No authors found'}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should render list of authors', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<MultiList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					renderNoResults={() => 'No authors found'}
					mockData={{
						'authors.keyword': {
							buckets: [
								{
									key: 'J. K. Rowling',
									doc_count: 10,
								},
								{
									key: 'Nora Roberts',
									doc_count: 7,
								},
							],
						},
					}}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should render search/count/checkbox when showSearch/showCount/showCheckbox are true', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<MultiList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					showSearch
					showCount
					showCheckbox
					renderNoResults={() => 'No authors found'}
					mockData={{
						'authors.keyword': {
							buckets: [
								{
									key: 'J. K. Rowling',
									doc_count: 10,
								},
								{
									key: 'Nora Roberts',
									doc_count: 7,
								},
							],
						},
					}}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should not render search/count/checkbox when showSearch/showCount/showCheckbox are false', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<MultiList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					showSearch={false}
					showCount={false}
					showCheckbox={false}
					renderNoResults={() => 'No authors found'}
					mockData={{
						'authors.keyword': {
							buckets: [
								{
									key: 'J. K. Rowling',
									doc_count: 10,
								},
								{
									key: 'Nora Roberts',
									doc_count: 7,
								},
							],
						},
					}}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should use renderItem to render the list item', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<MultiList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					renderNoResults={() => 'No authors found'}
					defaultValue={['J. K. Rowling']}
					renderItem={(key, docCount, isChecked) =>
						(isChecked ? (
							<div className="checked">
								{key} - {docCount}
							</div>
						) : (
							<div>
								{key} - {docCount}
							</div>
						))
					}
					mockData={{
						'authors.keyword': {
							buckets: [
								{
									key: 'J. K. Rowling',
									doc_count: 10,
								},
								{
									key: 'Nora Roberts',
									doc_count: 7,
								},
							],
						},
					}}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});

it('should use render prop to render the list item', () => {
	const elem = renderer
		.create(
			<ReactiveBase app="test" url="https://foo:bar@localhost:800">
				<MultiList
					mode="test"
					componentId="authors"
					dataField="authors.keyword"
					renderNoResults={() => 'No authors found'}
					mockData={{
						'authors.keyword': {
							buckets: [
								{
									key: 'J. K. Rowling',
									doc_count: 10,
								},
								{
									key: 'Nora Roberts',
									doc_count: 7,
								},
							],
						},
					}}
					render={({ data = [] }) => (
						<ul>
							{data.map(list => (
								<li key={list.key}>
									{list.key} {list.count}
								</li>
							))}
						</ul>
					)}
				/>
			</ReactiveBase>,
		)
		.toJSON();
	expect(elem).toMatchSnapshot();
});
