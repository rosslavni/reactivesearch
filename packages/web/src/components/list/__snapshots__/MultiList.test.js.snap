// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should not render search/count/checkbox when showSearch/showCount/showCheckbox are false 1`] = `
<div
  className="css-15g0dol-Base eay43c90"
  style={Object {}}
>
  <div
    className="css-a9jdxq-Container efy4jxw0"
    style={Object {}}
  >
    <ul
      aria-label="authors-items"
      className="css-1w1pnf0-UL e1y92esg2"
      role="listbox"
    >
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-12bsp2d-Checkbox-vh-hideInputControl e1y92esg1"
          id="authors-J. K. Rowling"
          name="authors-J. K. Rowling"
          onChange={[Function]}
          type="checkbox"
          value="J. K. Rowling"
        />
        <label
          className={null}
          htmlFor="authors-J. K. Rowling"
        >
          <span>
            <span>
              J. K. Rowling
            </span>
          </span>
        </label>
      </li>
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-12bsp2d-Checkbox-vh-hideInputControl e1y92esg1"
          id="authors-Nora Roberts"
          name="authors-Nora Roberts"
          onChange={[Function]}
          type="checkbox"
          value="Nora Roberts"
        />
        <label
          className={null}
          htmlFor="authors-Nora Roberts"
        >
          <span>
            <span>
              Nora Roberts
            </span>
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`should render list of authors 1`] = `
<div
  className="css-15g0dol-Base eay43c90"
  style={Object {}}
>
  <div
    className="css-a9jdxq-Container efy4jxw0"
    style={Object {}}
  >
    <input
      aria-label="authors-search"
      className="css-1f0ma1s-Input-input enqjda90"
      onChange={[Function]}
      placeholder="Search"
      style={
        Object {
          "margin": "0 0 8px",
        }
      }
      value=""
    />
    <ul
      aria-label="authors-items"
      className="css-1w1pnf0-UL e1y92esg2"
      role="listbox"
    >
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-t8nngi-Checkbox-vh e1y92esg1"
          id="authors-J. K. Rowling"
          name="authors-J. K. Rowling"
          onChange={[Function]}
          type="checkbox"
          value="J. K. Rowling"
        />
        <label
          className={null}
          htmlFor="authors-J. K. Rowling"
        >
          <span>
            <span>
              J. K. Rowling
            </span>
            <span
              className={null}
            >
              10
            </span>
          </span>
        </label>
      </li>
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-t8nngi-Checkbox-vh e1y92esg1"
          id="authors-Nora Roberts"
          name="authors-Nora Roberts"
          onChange={[Function]}
          type="checkbox"
          value="Nora Roberts"
        />
        <label
          className={null}
          htmlFor="authors-Nora Roberts"
        >
          <span>
            <span>
              Nora Roberts
            </span>
            <span
              className={null}
            >
              7
            </span>
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`should render no results message 1`] = `
<div
  className="css-15g0dol-Base eay43c90"
  style={Object {}}
>
  No authors found
</div>
`;

exports[`should render search/count/checkbox when showSearch/showCount/showCheckbox are true 1`] = `
<div
  className="css-15g0dol-Base eay43c90"
  style={Object {}}
>
  <div
    className="css-a9jdxq-Container efy4jxw0"
    style={Object {}}
  >
    <input
      aria-label="authors-search"
      className="css-1f0ma1s-Input-input enqjda90"
      onChange={[Function]}
      placeholder="Search"
      style={
        Object {
          "margin": "0 0 8px",
        }
      }
      value=""
    />
    <ul
      aria-label="authors-items"
      className="css-1w1pnf0-UL e1y92esg2"
      role="listbox"
    >
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-t8nngi-Checkbox-vh e1y92esg1"
          id="authors-J. K. Rowling"
          name="authors-J. K. Rowling"
          onChange={[Function]}
          type="checkbox"
          value="J. K. Rowling"
        />
        <label
          className={null}
          htmlFor="authors-J. K. Rowling"
        >
          <span>
            <span>
              J. K. Rowling
            </span>
            <span
              className={null}
            >
              10
            </span>
          </span>
        </label>
      </li>
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-t8nngi-Checkbox-vh e1y92esg1"
          id="authors-Nora Roberts"
          name="authors-Nora Roberts"
          onChange={[Function]}
          type="checkbox"
          value="Nora Roberts"
        />
        <label
          className={null}
          htmlFor="authors-Nora Roberts"
        >
          <span>
            <span>
              Nora Roberts
            </span>
            <span
              className={null}
            >
              7
            </span>
          </span>
        </label>
      </li>
    </ul>
  </div>
</div>
`;

exports[`should use render prop to render the list item 1`] = `
<div
  className="css-15g0dol-Base eay43c90"
  style={Object {}}
>
  <div
    className="css-a9jdxq-Container efy4jxw0"
    style={Object {}}
  >
    <input
      aria-label="authors-search"
      className="css-1f0ma1s-Input-input enqjda90"
      onChange={[Function]}
      placeholder="Search"
      style={
        Object {
          "margin": "0 0 8px",
        }
      }
      value=""
    />
    <ul>
      <li>
        J. K. Rowling
         
      </li>
      <li>
        Nora Roberts
         
      </li>
    </ul>
  </div>
</div>
`;

exports[`should use renderItem to render the list item 1`] = `
<div
  className="css-15g0dol-Base eay43c90"
  style={Object {}}
>
  <div
    className="css-a9jdxq-Container efy4jxw0"
    style={Object {}}
  >
    <input
      aria-label="authors-search"
      className="css-1f0ma1s-Input-input enqjda90"
      onChange={[Function]}
      placeholder="Search"
      style={
        Object {
          "margin": "0 0 8px",
        }
      }
      value=""
    />
    <ul
      aria-label="authors-items"
      className="css-1w1pnf0-UL e1y92esg2"
      role="listbox"
    >
      <li
        aria-checked={true}
        aria-selected={true}
        className="active"
        role="option"
      >
        <input
          checked={true}
          className="css-t8nngi-Checkbox-vh e1y92esg1"
          id="authors-J. K. Rowling"
          name="authors-J. K. Rowling"
          onChange={[Function]}
          type="checkbox"
          value="J. K. Rowling"
        />
        <label
          className={null}
          htmlFor="authors-J. K. Rowling"
        >
          <div
            className="checked"
          >
            J. K. Rowling
             - 
            10
          </div>
        </label>
      </li>
      <li
        aria-checked={false}
        aria-selected={false}
        className=""
        role="option"
      >
        <input
          checked={false}
          className="css-t8nngi-Checkbox-vh e1y92esg1"
          id="authors-Nora Roberts"
          name="authors-Nora Roberts"
          onChange={[Function]}
          type="checkbox"
          value="Nora Roberts"
        />
        <label
          className={null}
          htmlFor="authors-Nora Roberts"
        >
          <div>
            Nora Roberts
             - 
            7
          </div>
        </label>
      </li>
    </ul>
  </div>
</div>
`;
